.github/actions/setup-zeebe/action.yml

```diff
@@ -122,7 +119,7 @@ runs:
         -D maven.artifact.threads=32
         EOF
     - name: Cache local Maven repository
-      if: inputs.java == 'true' && startsWith(runner.name, 'actions-runner-')
+      if: inputs.java == 'true' && (startsWith(runner.name, 'actions-runner-') || startsWith(runner.name, 'aws-') || startsWith(runner.name, 'gcp-'))
```

Comment:
I don't really like this, do you think it's worth extracting the logic to another job where we could make use of regex and output a simple selfHosted boolean flag to be used in the conditions? 🤔

Reply:
```
```
+ [[ aws-arm-core-8-default =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ gcp-core-2-default =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ n1-standard-8-netssd-preempt-quick =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=self-hosted
result=self-hosted

+ [[ whatever =~ ^(actions-runner-|gcp-|aws-|n1-).*$ ]]
+ echo result=gh-hosted
result=gh-hosted
```